content/dev/plugins/encapsulated_plugins/converting.md

```diff
@@ -67,3 +67,27 @@ In our example, these will be placed in
 ./zc_plugins/ModList/1.4.0/Installer/install.sql
 ```
 
+### 5. Things to watch for in the conversion 
+
+a) Scope of variables in `extra_configures` and `extra_datafiles`
+
+In the encapsulated plugin architecture, 
+the files in `extra_configures` and `extra_datafiles` are run in the 
+context of  the `FileSystem` class, 
+and therefore any variables created will be scoped into that class and not the global scope.
+
+This means if you have a file that was in `admin/includes/extra_configures/my_plugin.php` which created a variable (say, `$my_list`), this variable will not be available to the plugin anymore.  
+
+Some options for overcoming this are: 
+
+- Move the logic to a page specific init file. 
+
+- Make the variables into defined constants (which by definition have global scope). 
+
+### 6. Remove old files and update documentation 
+
+Please *do not* preserve top level folders like `admin` and `includes` with the same files you have created in `zc_plugins/YOURPLUGIN/VERSION/`.  Instead, update the README file for your plugin with guidance like: 
+
```

Comment:
> do not preserve top level folders like `admin` and `includes`

"do not preserve these folders" seems confusing, as one might interpret it as "under zc_plugins don't create any directories named "admin" or "includes". But that's contradictory to what does need to be done.

Did you mean "don't just copy all your old-style files into zc_plugins and call it a day; instead, review every file and ensure its relevance, suitability, and proper location and function, rewriting whatever parts are necessary to conform to the new requirements, and taking the time to rewrite things that could be modernized, and removing things that are just carryovers of old ways of doing things"? or something to that effect?

Reply:
Not exactly ... I meant "don't replicate them at the top level as well."   Let me rework some of this language.